mountpoint-s3/src/fs.rs

```diff
@@ -1,3 +1,31 @@
+//! File system types and operations, not relying on our FUSE library.
+//!
+//! **Note:** this abstraction is not intended to abstract FUSE itself,
+//! only allow us to run tests without FUSE in the loop.
+//! If and when we add a new filesystem view that isn't FUSE related,
+//! we can consider how to ensure this module is generic.
+//!
+//! #### Counting references to [Inode]
+//!
+//! One of the responsibilities of [S3Filesystem] is to track the _lookup count_.
+//! Lookup count is a FUSE concept and tracks the number of times we provide a reference to an [InodeNo] to the FUSE driver,
+//! ensuring that we never delete an [Inode] that may be references in the future by the FUSE driver.
+//! Note - this should not be confused with any counting of references to an [Inode] from its parent or a [FileHandle].
+//!
+//! Some operations will not return a new [InodeNo] but rather one the FUSE driver already knows about.
+//! `open` is one such case since FUSE already provides the [InodeNo] as an argument when it calls open.
```

Comment:
This isn't quite the intent -- any operation that takes an InodeNo as _input_ doesn't need to change the lookup count of that `InodeNo`.